drivers/i2c/sun6i-a31-i2c.c

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright © 2017-2018 The Crust Firmware Authors.
+ * SPDX-License-Identifier: (BSD-3-Clause OR GPL-2.0)
+ */
+
+#include <clock.h>
+#include <dm.h>
+#include <error.h>
+#include <i2c.h>
+#include <mmio.h>
+#include <stddef.h>
+#include <util.h>
+#include <wallclock.h>
+
+#define I2C_CTRL_REG 0x0c
+#define I2C_DATA_REG 0x08
+#define I2C_STAT_REG 0x10
+
+#define I2C_TIMEOUT  10 /*< Timeout in useconds. */
+
+enum I2CStatus {
+	START_COND_TX        = 0x08,
+	START_COND_TX_REPEAT = 0x10,
+	ADDR_WRITE_RX_ACK    = 0x18,
+	ADDR_WRITE_RX_NACK   = 0x20,
+	DATA_TX_ACK          = 0x28,
+	DATA_TX_NACK         = 0x30,
+	ADDR_READ_TX_ACK     = 0x40,
+	ADDR_READ_RX_ACK     = 0x48,
+	DATA_RX_NACK         = 0x58,
+	IDLE                 = 0xf8,
+};
+
+static bool sun6i_a31_i2c_verify_status(struct device *dev,
+                                        enum I2CStatus status);
+static int sun6i_a31_i2c_write(struct device *dev, uint8_t data);
+
+static int
+sun6i_a31_i2c_read(struct device *dev, uint8_t *data)
+{
+	/* Wait for data. */
+	if (!sun6i_a31_i2c_verify_status(dev, DATA_RX_NACK))
+		return ENODEV;
+
+	/* Read data. */
+	*data = mmio_read32(dev->regs + I2C_DATA_REG) & BITMASK(0, 7);
```

Comment:
C truncates to the destination size for you, no need to mask

Reply:
Removed.